### ISSUE TITLE
Question about the generalization capability of a BNN

### ISSUE THREAD

**Issue by ben-arnao** [tensorflow/probability#827]
I know that one of the main benefits of a BNN is the ability to gauge the confidence of a prediction. I understand why this can be important for various applications but i am more concerned about the advantages of BNN's ability to generalize over a traditional NN with the right selection of generalization techniques (batch norm, dropout, regularization, etc.).

Specifically i am wondering from a generalization perspective why BNN is better than just adding gaussian noise to our modeL?

**Comment by nbro** [comment]
@ben-arnao Bayesian neural networks are arguably a more principled way of tackling uncertainty than dropout, etc. In terms of generalization, I would also like to see more theoretical research work on the topic. Maybe this would be a good research topic.
